File: app/src/main/java/com/ivanmagda/habito/view/model/HabitDetailViewModel.java
package com.ivanmagda.habito.view.model;

import android.support.annotation.NonNull;

import com.ivanmagda.habito.barchart.HabitoBarChartRange;
import com.ivanmagda.habito.utils.HabitoDateUtils;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public final class HabitDetailViewModel {

    private static final SimpleDateFormat WEEK_FORMAT = new SimpleDateFormat("d MMM yyyy",
            Locale.getDefault());
    private static final SimpleDateFormat MONTH_FORMAT = new SimpleDateFormat("d MMM yyyy",
            Locale.getDefault());
    private static final SimpleDateFormat YEAR_FORMAT = new SimpleDateFormat("MMM yyyy",
            Locale.getDefault());

    private HabitoBarChartRange.DateRange mDateRange = HabitoBarChartRange.DateRange.WEEK;

    public HabitDetailViewModel() {
    }

    public HabitDetailViewModel(@NonNull HabitoBarChartRange.DateRange dateRange) {
        this.mDateRange = dateRange;
    }

    public void setDateRange(HabitoBarChartRange.DateRange dateRange) {
        this.mDateRange = dateRange;
    }

    public String getScoreString(int score) {
        return String.valueOf(score);
    }

    public String getDateRangeString() {
        switch (mDateRange) {
            case WEEK:
                return getFormattedWeek(HabitoDateUtils.getStartOfCurrentWeek(),
                        HabitoDateUtils.getEndOfCurrentWeek());
            case MONTH:
                return getFormattedMonth(HabitoDateUtils.getStartOfCurrentMonth(),
                        HabitoDateUtils.getEndOfCurrentMonth());
            case YEAR:
                return getFormattedYear(HabitoDateUtils.getStartOfCurrentYear(),
                        HabitoDateUtils.getEndOfCurrentYear());
            default:
                throw new IllegalArgumentException("Receive illegal date range");
        }
    }

    private String getFormattedWeek(long start, long end) {
        return formatDates(WEEK_FORMAT, start, end);
    }

    private String getFormattedMonth(long start, long end) {
        return formatDates(MONTH_FORMAT, start, end);
    }

    private String getFormattedYear(long start, long end) {
        return formatDates(YEAR_FORMAT, start, end);
    }

    private String formatDates(SimpleDateFormat dateFormat, long start, long end) {
        return dateFormat.format(new Date(start))
                + " - "
                + dateFormat.format(new Date(end));
    }

}

File: app/src/main/java/com/ivanmagda/habito/view/model/HabitDetailViewModel.kt
package com.ivanmagda.habito.view.model

import com.ivanmagda.habito.barchart.HabitoBarChartRange
import com.ivanmagda.habito.util.HabitoDateUtils
import java.text.SimpleDateFormat
import java.util.*

class HabitDetailViewModel(var dateRange: HabitoBarChartRange.DateRange = HabitoBarChartRange.DateRange.WEEK) {

    val dateRangeString: String
        get() {
            return when (dateRange) {
                HabitoBarChartRange.DateRange.WEEK -> getFormattedWeek(HabitoDateUtils.startOfCurrentWeek,
                        HabitoDateUtils.endOfCurrentWeek)
                HabitoBarChartRange.DateRange.MONTH -> getFormattedMonth(HabitoDateUtils.startOfCurrentMonth,
                        HabitoDateUtils.endOfCurrentMonth)
                HabitoBarChartRange.DateRange.YEAR -> getFormattedYear(HabitoDateUtils.startOfCurrentYear,
                        HabitoDateUtils.endOfCurrentYear)
            }
        }

    fun getScoreString(score: Int): String {
        return score.toString()
    }

    private fun getFormattedWeek(start: Long, end: Long): String {
        return formatDates(WEEK_FORMAT, start, end)
    }

    private fun getFormattedMonth(start: Long, end: Long): String {
        return formatDates(MONTH_FORMAT, start, end)
    }

    private fun getFormattedYear(start: Long, end: Long): String {
        return formatDates(YEAR_FORMAT, start, end)
    }

    private fun formatDates(dateFormat: SimpleDateFormat, start: Long, end: Long): String {
        return (dateFormat.format(Date(start))
                + " - "
                + dateFormat.format(Date(end)))
    }

    companion object {
        private val WEEK_FORMAT = SimpleDateFormat("d MMM yyyy", Locale.getDefault())
        private val MONTH_FORMAT = SimpleDateFormat("d MMM yyyy", Locale.getDefault())
        private val YEAR_FORMAT = SimpleDateFormat("MMM yyyy", Locale.getDefault())
    }
}
